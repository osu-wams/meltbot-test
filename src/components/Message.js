import React from 'react';
import styled from 'styled-components';

const StyledMessage = styled.div`
  max-width: 80%;
  background-color: #d73f09;
  border-radius: 20px;
  padding: 16px;
  margin-bottom: 12px;
  ${props => {
    if (props.type === 'bot') {
      return `
    background-color: #ffffff;
    border-radius: 16px 16px 16px 0;
    align-self: flex-start;
  `;
    }
    if (props.type === 'user') {
      return `
    border-radius: 16px 16px 0 16px;
    align-self: flex-end;
    color: #ffffff;
  `;
    }
  }}
`;

const Message = ({ type, ...props }) => {
  return <StyledMessage type={type} {...props} />;
};

export default Message;
