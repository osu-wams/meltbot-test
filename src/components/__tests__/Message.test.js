import React from 'react';
import { render } from '@testing-library/react';
import Message from '../Message';

const BotMessage = () => (
  <Message type="bot">
    <p>Bot Message</p>
  </Message>
);

const UserMessage = () => (
  <Message type="user">
    <p>User Message</p>
  </Message>
);

const FollowUpQuestion = () => (
  <Message>
    <p>Follow up question button</p>
  </Message>
);

test('BotMessage - Message with type="bot" is styled appropriately', () => {
  const { container } = render(<BotMessage />);
  expect(container).toMatchSnapshot();
});

test('UserMessage - Message with type="user" is styled appropriately', () => {
  const { container } = render(<UserMessage />);
  expect(container).toMatchSnapshot();
});

test('FollowUpQuestion - Message with no type - is styled appropriately', () => {
  const { container } = render(<FollowUpQuestion />);
  expect(container).toMatchSnapshot();
});
