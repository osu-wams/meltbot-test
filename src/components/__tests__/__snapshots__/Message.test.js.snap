// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotMessage - Message with type="bot" is styled appropriately 1`] = `
.c0 {
  display: inline-block;
  max-width: 70%;
  background-color: #d73f09;
  border-radius: 20px;
  padding: 16px;
  margin-bottom: 12px;
  text-align: left;
  background-color: #ffffff;
  border-radius: 16px 16px 16px 0;
  position: relative;
  margin-left: 60px;
  margin-bottom: 30px;
  border: 1px solid #B7B1AF;
}

.c0 > span + p {
  margin-top: 0;
}

.c0 > :last-child {
  margin-bottom: 0;
}

.c0:before {
  content: ' ';
  background: url(benny.png) no-repeat center center;
  width: 50px;
  height: 50px;
  position: absolute;
  left: -60px;
  bottom: -20px;
}

.c0 a {
  color: #D73F09;
}

<div>
  <div
    class="c0"
    data-testid="message"
    type="bot"
  >
    <p>
      Bot Message
    </p>
  </div>
</div>
`;

exports[`FollowUpQuestion - Message with no type - is styled appropriately 1`] = `
.c0 {
  display: inline-block;
  max-width: 70%;
  background-color: #d73f09;
  border-radius: 20px;
  padding: 16px;
  margin-bottom: 12px;
  text-align: left;
}

.c0 > span + p {
  margin-top: 0;
}

.c0 > :last-child {
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
    data-testid="message"
  >
    <p>
      Follow up question button
    </p>
  </div>
</div>
`;

exports[`UserMessage - Message with type="user" is styled appropriately 1`] = `
.c0 {
  display: inline-block;
  max-width: 70%;
  background-color: #d73f09;
  border-radius: 20px;
  padding: 16px;
  margin-bottom: 12px;
  text-align: left;
  right: 0;
  border-radius: 16px 16px 0 16px;
  color: #ffffff;
}

.c0 > span + p {
  margin-top: 0;
}

.c0 > :last-child {
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
    data-testid="message"
    type="user"
  >
    <p>
      User Message
    </p>
  </div>
</div>
`;
